include/spdlog/details/os-inl.h

```diff
@@ -484,15 +484,15 @@ SPDLOG_INLINE void utf8_to_wstrbuf(string_view_t str, memory_buf_t &target)
     }
 
     int result_size = static_cast<int>(target.capacity());
-    if ((str_size + 1) * 2 < result_size)
+    if (str_size + 1 > result_size)
     {
         result_size = ::MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, str.data(), str_size, NULL, 0);
     }
 
     if (result_size > 0)
     {
         target.resize(result_size);
-        result_size = ::MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, str.data(), str_size, (LPWSTR)target.data(), result_size);
+        result_size = ::MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, str.data(), str_size, target.data(), result_size);
 
```

Comment:
Why remove the ```(LPWSTR)``` cast ? Does  ```target.data()``` guaranteed to be (LPWSTR) ?

Reply:
Yes, because now it uses a `wmemory_buffer`, which as the w prefix says, is always wide.